### DISCUSSION TITLE
build self contained theme folder

### DISCUSSION THREAD

**Discussion by setop** [thuliteio/doks#698] [Ideas]
As of now, when building (`npm run build`) the theme assets (archetypes, data, i18n, images, layouts,  static and  theme.toml)  are mixed with the content of the repo making it difficult to use it in a classic hugo project (obtain with `hugo new site foo`).

It there a proper way to collect these asset in a separated folder so it can be move elsewhere ?

**Comment by h-enk** [comment]
It's not in the planning to release Doks as a submodule or module. I tried that before, but Hugo does not support a npm based theme setup (mainly the scripts part) — yet.

**Comment by doganulus** [comment]
One reason I prefer Hugo to the other alternatives is the possibility to avoid the `node_modules` menace. Recently docsy-theme has switched to [Go-module-based installation](https://github.com/google/docsy/issues/849) and I think it is the way for Hugo themes. The `npm` does not really fit into the whole picture in my honest opinion.
